web_src/js/features/issuesutil.js

```diff
@@ -0,0 +1,1471 @@
+}
+
+// subscribe-repo
+// toggle_stopwatch_form"
+// cancel_stopwatch_form"
+// add_time_manual_form
+// addDependencyForm
+function initAjaxForms(formId) {
+  $(formId).submit(function(e) {
+    e.preventDefault(); // avoid to execute the actual submit of the form.
+
+    const form = $(this);
+    const url = form.attr('action');
+
+    $.ajax({
+      type: 'POST',
+      url,
+      data: form.serialize(), // serializes the form's elements.
+      success(_data) {
+        $('#lock').modal('hide');
+        $('#lock').remove();
+        reloadIssuesActions();
+      }
+    });
+  });
+}
+
+export function initIssueList() {
+  const repolink = $('#repolink').val();
+  const repoId = $('#repoId').val();
+  const crossRepoSearch = $('#crossRepoSearch').val();
+  const tp = $('#type').val();
+  let issueSearchUrl = `${AppSubUrl}/api/v1/repos/${repolink}/issues?q={query}&type=${tp}`;
+  if (crossRepoSearch === 'true') {
+    issueSearchUrl = `${AppSubUrl}/api/v1/repos/issues/search?q={query}&priority_repo_id=${repoId}&type=${tp}`;
+  }
+  $('#new-dependency-drop-list')
+    .dropdown({
+      apiSettings: {
+        url: issueSearchUrl,
+        onResponse(response) {
+          const filteredResponse = {success: true, results: []};
+          const currIssueId = $('#new-dependency-drop-list').data('issue-id');
+          // Parse the response from the api to work with our dropdown
+          $.each(response, (_i, issue) => {
+            // Don't list current issue in the dependency list.
+            if (issue.id === currIssueId) {
+              return;
+            }
+            filteredResponse.results.push({
+              name: `#${issue.number} ${htmlEscape(issue.title)
+              }<div class="text small dont-break-out">${htmlEscape(issue.repository.full_name)}</div>`,
+              value: issue.id
+            });
+          });
+          return filteredResponse;
+        },
+        cache: false,
+      },
+
+      fullTextSearch: true
+    });
+
+  $('.menu a.label-filter-item').each(function () {
+    $(this).on('click', function (e) {
+      if (e.altKey) {
+        e.preventDefault();
+
+        const href = $(this).attr('href');
+        const id = $(this).data('label-id');
+
+        const regStr = `labels=(-?[0-9]+%2c)*(${id})(%2c-?[0-9]+)*&`;
+        const newStr = 'labels=$1-$2$3&';
+
+        window.location = href.replace(new RegExp(regStr), newStr);
+      }
+    });
+  });
+
+  $('.menu .ui.dropdown.label-filter').on('keydown', (e) => {
+    if (e.altKey && e.keyCode === 13) {
+      const selectedItems = $('.menu .ui.dropdown.label-filter .menu .item.selected');
+
+      if (selectedItems.length > 0) {
+        const item = $(selectedItems[0]);
+
+        const href = item.attr('href');
+        const id = item.data('label-id');
+
+        const regStr = `labels=(-?[0-9]+%2c)*(${id})(%2c-?[0-9]+)*&`;
+        const newStr = 'labels=$1-$2$3&';
+
+        window.location = href.replace(new RegExp(regStr), newStr);
```

Comment:
This code is duplicated, I'd extract it to a function, e.g. `getNewLocation(href, labelId)`.

Reply:
this was there before my PR, I'll leave to someone who understands it better.